packages/driver/src/dom/visibility/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,330 @@
+# Fast Visibility Algorithm Migration Guide
+
```

Comment:
@cacieprins Been meaning to ask how much you see portions of this being in the https://docs.cypress.io/app/references/experiments page vs here. I was thinking there could be some basic explanation in the docs and then link to here since it's quite extensive.